redbeams/src/main/java/com/sequenceiq/redbeams/service/dbserverconfig/DatabaseServerConfigService.java

```diff
@@ -23,4 +32,74 @@
         return repository.findAllByWorkspaceIdAndEnvironmentId(workspaceId, environmentId);
     }
 
+    public DatabaseServerConfig create(DatabaseServerConfig resource, Long workspaceId) {
+        // FIXME? Currently no checks if logged-in user has access to workspace
+        // Compare with AbstractWorkspaceAwareResourceService
+        try {
+            MDCBuilder.buildMdcContext(resource);
+            // prepareCreation(resource);
+            resource.setWorkspaceId(workspaceId);
+            return repository.save(resource);
+        } catch (AccessDeniedException e) {
+            ConstraintViolationException cve = null;
+            for (Throwable t = e.getCause(); t != null; t = t.getCause()) {
+                if (t instanceof ConstraintViolationException) {
+                    cve = (ConstraintViolationException) t;
+                    break;
+                }
+            }
+            if (cve != null) {
+                String message = String.format("%s already exists with name '%s' in workspace %d",
+                        resource().getShortName(), resource.getName(), resource.getWorkspaceId());
+                throw new BadRequestException(message, e);
+            }
+            throw e;
+        }
+    }
+
+    public DatabaseServerConfig getByNameInWorkspace(Long workspaceId, String name) {
+        Optional<DatabaseServerConfig> resourceOpt = repository.findByNameAndWorkspaceId(name, workspaceId);
+        if (resourceOpt.isEmpty()) {
+            throw new NotFoundException(String.format("No %s found with name '%s'", resource().getShortName(), name));
+        }
+        MDCBuilder.buildMdcContext(resourceOpt.get());
+        return resourceOpt.get();
+    }
+
+    public DatabaseServerConfig deleteByNameInWorkspace(Long workspaceId, String name) {
+        DatabaseServerConfig resource = getByNameInWorkspace(workspaceId, name);
+        return delete(resource);
+    }
+
+    DatabaseServerConfig delete(DatabaseServerConfig resource) {
+        LOGGER.debug("Deleting {} with name: {}", resource().getReadableName(), resource.getName());
+        MDCBuilder.buildMdcContext(resource);
+        // prepareDeletion(resource);
+        repository.delete(resource);
```

Comment:
DaabaseServerConfig was marked as ArchivableResource. We should decide if it makes sense to do a soft or hard delete on it. We can, however treat archiving as a separate PR - I will also check what kind of refactoring will happen to the archiving machinery in cloudbreak.